prometheus-to-sd/config/dynamic_source.go

```diff
@@ -31,11 +50,8 @@ func SourceConfigsFromDynamicSources(gceConfig *GceConfig, sources []flags.Uri)
 	if err != nil {
 		return nil, err
 	}
-	podResponse, err := kubeApi.CoreV1().Pods(podNamespace).List(createOptionsForPodSelection(gceConfig.Instance, sourceMap))
```

Comment:
Please correct me if I missed some part of your code, but how do you get an initial list of pods? Do you get from watch the events that happened before prom-to-sd creation?

Reply:
This is what we discovered with Marian last week, and discussed with Marek about yesterday. When you start the watch, you will get an ADDED event for all the pods that are already started and working, so this way you do not need to query for the initial pod list